services/blockchain-connector/shared/database/better-sqlite3.js

```diff
@@ -0,0 +1,597 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const fs = require('fs');
+const { Logger } = require('lisk-service-framework');
+const config = require('../../config');
+
+const logger = Logger();
+
+const connectionPool = {};
+const tablePool = {};
+const SQLITE3_DB_FILE_EXTENSION = 'sqlite3';
+
+const loadSchema = async (knex, tableName, tableConfig) => {
+	const { primaryKey, charset, schema, indexes } = tableConfig;
+
+	if (await knex.schema.hasTable(tableName)) return knex;
+
+	await knex.schema
+		.createTable(tableName, table => {
+			if (charset) table.charset(charset);
+
+			Object.keys(schema).map(p => {
+				const kProp = (table[schema[p].type])(p);
+				if (schema[p].null === false) kProp.notNullable();
+				if ('defaultValue' in schema[p]) kProp.defaultTo(schema[p].defaultValue);
+				if (indexes[p]) kProp.index();
+				return kProp;
+			});
+			table.primary(primaryKey);
+		});
+
+	return knex;
+};
+
+const createDbConnection = async (dbDataDir, tableName) => {
+	const knex = require('knex')({
+		client: 'better-sqlite3',
+		connection: {
+			filename: `./${dbDataDir}/${tableName}_db.${SQLITE3_DB_FILE_EXTENSION}`,
+		},
+		useNullAsDefault: true,
+		pool: {
+			max: 100,
+			min: 2,
+		},
+		log: {
+			warn(message) { logger.warn(message); },
+			error(message) { logger.error(message); },
+			deprecate(message) { logger.warn(message); },
+			debug(message) { logger.debug(message); },
+		},
+	});
+
+	knex.select(1)
+		.on('query-error', error => {
+			logger.error(error.message);
+		})
+		.catch(err => {
+			if (err.code === 'ECONNREFUSED') {
+				logger.error(err.message);
+				logger.fatal('Unable to connect to the database, shutting down the process...');
+				process.exit(1);
+			}
+			logger.error(err.message);
+		});
+
+	return knex;
+};
+
+const cast = (val, type) => {
```

Comment:
Do we need to handle following conditions here?

1. `val` is `undefined`
2. `type` is `range`

Reply:
Not required. The implementation must be kept in sync with the MySQL implementation.